Makefile

```diff
@@ -53,7 +53,7 @@ include mshadow/make/mshadow.mk
 include $(DMLC_CORE)/make/dmlc.mk
 
 # all tge possible warning tread
-WARNFLAGS= -Wall -Wsign-compare
+WARNFLAGS= -Wall -Wsign-compare -Werror
```

Comment:
This is something that would need to be discussed on dev.  Also, when developing, you don't want to have errors on warnings